provider/pkg/provider/webhook.go

```diff
@@ -265,42 +265,43 @@ func (wh *PulumiServiceWebhookResource) Diff(req *pulumirpc.DiffRequest) (*pulum
 		}
 	}
 
-	changes := pulumirpc.DiffResponse_DIFF_NONE
-	var diffs, replaces []string
-	properties := map[string]bool{
-		"active":           false,
-		"displayName":      false,
-		"payloadUrl":       false,
-		"secret":           false,
-		"format":           false,
-		"filters":          false,
+	diffs := olds.Diff(news)
+	if diffs == nil {
+		return &pulumirpc.DiffResponse{
+			Changes: pulumirpc.DiffResponse_DIFF_NONE,
+		}, nil
+	}
+
+	dd := plugin.NewDetailedDiffFromObjectDiff(diffs)
+
+	detailedDiffs := map[string]*pulumirpc.PropertyDiff{}
+	replaceProperties := map[string]bool{
 		"organizationName": true,
 		"projectName":      true,
 		"stackName":        true,
 	}
-	if d := olds.Diff(news); d != nil {
-		for key, replace := range properties {
-			i := sort.SearchStrings(req.IgnoreChanges, key)
-			if i < len(req.IgnoreChanges) && req.IgnoreChanges[i] == key {
-				continue
-			}
-
-			if d.Changed(resource.PropertyKey(key)) {
-				changes = pulumirpc.DiffResponse_DIFF_SOME
-				diffs = append(diffs, key)
-
-				if replace {
-					replaces = append(replaces, key)
-				}
-			}
+	for k, v := range dd {
+		// we don't care about the name property because it is only an output
+		if v.Kind == 2 && k == "name" {
+			continue
+		}
```

Comment:
If I don't do this, the `name` property *always* shows up as a `Delete` and the resource *always* shows an `Update`. I feel like I shouldn't have to do this because it's defined in the schema that `name` is just an `input` and not an `output`. Any ideas what I'm doing wrong here?

Reply:
I don't think that you're doing anything wrong. I think that this is because `olds` and `news` are shaped differently and the differ is not schema-aware. You could avoid this by preprocessing `news` to add a matching `name` property or pre-processing `olds` to remove the `name` property.